src/components/modules/paste.ts

```diff
@@ -195,10 +195,17 @@ export default class Paste extends Module {
       } catch (e) { } // Do nothing and continue execution as usual if error appears
     }
 
-    // Creates an <a> tag for links within plain texts
-    htmlData = plainData.split(` `).map((word) => {
-      return ((word.startsWith(`http:`)) || (word.startsWith(`https:`))) ? `<a href="${word}">${word}</a>` : word;
-    }).join(` `);
+    // Creates an <a> tag for links within plain text
+    const links = plainData.split(' ')
+      .map(word => word.split('\n'))
+      .flat()
+      .filter((word) => {
+        return ((word.startsWith('http:')) || (word.startsWith('https:')));
+      });
+
+    for (const link of links) {
+      htmlData = htmlData.replace(link, `<a href="${link}">${link}</a>`);
+    }
```

Comment:
Issue here: if the same link exists twice in the pasted text you will be replacing the `href` text again like `<a href="<a href="link">link</a>">link</a>`.

It also seems to get messed up by HTML anchor tags that are pasted in with existing links (i.e. paste the same hyperlink from google docs twice).

https://github.com/discolearn/editor.js/assets/33005811/3db995ef-54fb-4eaf-ab22-2d1b2425e639

Reply:
Ah good catch, made it so we ignore any links that are prefixed by `href="` or followed by `</a>`

https://github.com/discolearn/editor.js/assets/41494387/d53ea17f-c3e9-4dc3-af21-ff090229168c